test/suites/optional/observability.robot

```diff
@@ -0,0 +1,63 @@
+*** Settings ***
+Documentation       This suite performs basic log scans to determine if the opentelemetry-collector is exporting data
+...                 and in a healthy state
+
+Library             OperatingSystem
+Library             String
+Library             ../../resources/journalctl.py
+Resource            ../../resources/kubeconfig.resource
+Resource            ../../resources/common.resource
+
+Suite Setup         Setup Suite And Set Journal Cursor
+Suite Teardown      Teardown Suite
+
+
+*** Variables ***
+${JOURNAL_CUR}      ${EMPTY}
+
+
+*** Test Cases ***
+Logs Should Contain Exported Metric Data
+    [Documentation]    OpenTelemetry logs should contain the data from the LogsExporter and MetricsExporter
+
+    ${pattern}    Catenate    SEPARATOR=
+    ...    info\\s+MetricsExporter\\s+{"kind": "exporter", "data_type": "metrics", "name": "debug", "resource metrics":
+    ...    \ [0-9]+, "metrics": [0-9]+, "data points": [0-9]+}
+    Pattern Should Appear In Log Output    ${JOURNAL_CUR}    ${pattern}    unit="microshift-observability"
+
+Logs Should Contain Exported Log Data
+    [Documentation]    OpenTelemetry logs should not contain errors by any component (reciver, processor,exporter).
+
+    ${pattern}    Catenate
+    ...    SEPARATOR=
+    ...    info\\s+LogsExporter\\s+\\{"kind": "exporter", "data_type": "logs", "name": "debug", "resource logs": [0-9]+,
+    ...    \ "log records": [0-9]+\\}
+    Pattern Should Appear In Log Output    ${JOURNAL_CUR}    ${pattern}    unit="microshift-observability"
+
+Logs Should Not Contain Reciever Errors
+    [Documentation]    Internal receiver errors are not treated as fatal. Typically these are due to a miconfiguration
+    ...    and thus indicate the provided default config should be reviewed.
+
+    ${pattern}    Catenate    SEPARATOR= \\s+\\{"error":.*\\}
+    Pattern Should Not Appear In Log Output    ${JOURNAL_CUR}    ${pattern}    unit="microshift-observability"
+
+
+*** Keywords ***
+Setup Suite And Set Journal Cursor
+    [Documentation]    The service starts after MicroShift starts and thus will start generating pertinant log data
+    ...    right away. When the suite is executed, immediately get the cursor for the current
+    Setup Suite
+    ${cur}    Get Journal Cursor
```

Comment:
Should we specify a unit here?

Reply:
The missing unit was an oversight on my part. Fixed now.  I don't understand what you mean about the race condition though. Can you elaborate?